[MTDB-12811] - feat(leaderboard): :sparkles: Adds Leaderboard view, removes Leaderboard from sidebar folder

ssotomayor opened the pull request:
- Leaderboard view :sparkles:
- Adds react-router :sparkles:

PorlyBe commented on src/App.tsx:
```diff
@@ -90,7 +103,7 @@ const App: FC = () => {
       {status !== "connected" || loading || !isSocketConnected ? (
         <NotConnected />
```
lets move this and the logic into Main so users can view leaderboard without being connected

PorlyBe commented on src/components/Header/styled.ts:
```diff
@@ -34,8 +34,12 @@ export const LogoWrapper = styled.a`
 export const Title = styled(H3)`
   grid-area: title;
   text-align: left;
+  @media (min-width: 0px) {
```
you shouldn't need this media query.. also `vw` for fontsizes. yikes, lol

ssotomayor replied:
Oh i needed it because i added 
``` 
<Title>
        Verse {gameMode === "Christmas" ? "Clickmas" : "Clicker"}{" "}
        {pageTitle ? `| ${pageTitle}` : ""}
</Title>
```

And then was passing:
```
<Header pageTitle="Leaderboard" />
```

Specifically for this component, this made the title "Verse Clicker | Leaderboard" which was longer and breaking a bit in phones, i've removed the title now, but i thought it was a nice touch :D

PorlyBe commented on src/components/Sidebar/Sidebar.tsx:
```diff
@@ -50,8 +45,7 @@ const Sidebar = () => {
           onClick={() => {
-            setContent("LEADERBOARD");
-            showModal();
+            navigate("/leaderboard");
```
🤔 thinking lets keep the modal for ingame quick view... navigating back can be a litlte janky when viewed inside verse explorer

PorlyBe reviewed (changes requested):
Finally got around to reviewing this... left some comments, also some more in here

- please rebase onto master
- particles are visible in front of leaderboard
- i think we should separate the components from current (modal) leaderbaord and this page, since they'll have different designs (once we get designers to whip something up)

ssotomayor commented:
@PorlyBe changes are done, regarding this:
> i think we should separate the components from current (modal) leaderbaord and this page, since they'll have different designs (once we get designers to whip something up)

Components from the specific view are in _src/components/Leaderboard_ and for the Sidebar are within _src/components/Sidebar/Leaderboard_

I understand this can be confusing, but it felt weird to call the folder "LeaderboardView" within "components" open to suggestions tho. 🤔

PorlyBe commented on src/views/Leaderboard.tsx:
```diff
@@ -0,0 +1,70 @@
+  const GAME_MODES: Array<{ label: string; value: GameMode }> = [
+    { label: "Ethereum", value: "Ethereum" },
+    { label: "Polygon", value: "Polygon" },
+    { label: "Goerli", value: "Goerli" },
```
can we remove testnets from this if REACT_APP_DEV_ENV=development ??

ssotomayor replied:
Done.

PorlyBe reviewed (approved):
one small comment but looks good

micahpatng reviewed (commented):
I have a few points that I wanna discuss:
- the particles are seen behind the campaign tabs
- maybe we should have a view of the leaderboard when there are no values at all in the campaign? 
- would be nice to have a padding for the leaderboard for mobile
![File (2)](https://github.com/bitcoin-verse/verse-clicker/assets/94164690/f6215c29-e698-4933-9008-aede3d4f3558)

ssotomayor commented:
@micahpatng what you think? I fixed the particles going through the campaign, made the leaderboard slightly smaller to add gutter on mobile, and changed it to be like this when empty

<img width="364" alt="Screenshot 2024-02-21 at 01 12 47" src="https://github.com/bitcoin-verse/verse-clicker/assets/1304919/f2c6f38b-8e40-4447-9160-0544dfea53a3">

micahpatng commented on src/components/Leaderboard/LeaderboardViewer.tsx:
```diff
@@ -0,0 +1,98 @@
+      <LeaderboardContent>
+        {leaderboardItems.length ? (
+          leaderboardItems?.map(({ address: addr, stats }, index) => {
+            if (index >= 25) {
```
`if (index >= 25 && addr === address)` maybe?

ssotomayor replied:
This logic was copied from:
https://github.com/bitcoin-verse/verse-clicker/blob/1b274e1071bad62137f05bcc1501e0dbbbf2c333/src/components/Sidebar/Leaderboard/Leaderboard.tsx#L54-L68

Which seems like only if the user wallet is above the 25th rank it should be shown, if not it doesn't show the row at all.

I've refactored it now in both components. I wouldn't change much of it atm since it will most likely be changed when the designers spin up something.

micahpatng commented on src/components/Leaderboard/LeaderboardViewer.tsx:
```diff
@@ -0,0 +1,98 @@
+              />
+            );
+          })
+        ) : (
```
also would be better to have this into separate components so it's more readable? like `{leaderboardItems.length ? </LeaderboardConent> :</EmptyLeaderboard>}`

ssotomayor replied:
Done.

micahpatng reviewed (approved):
lgtm, merge it in!
